x-pack/plugins/siem/public/components/flow_controls/flow_target_select.test.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mount, shallow } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import { clone } from 'lodash/fp';
+import * as React from 'react';
+import { ActionCreator } from 'typescript-fsa';
+
+import { FlowDirection, FlowTarget } from '../../graphql/types';
+
+import { FlowTargetSelect } from './flow_target_select';
+
+describe('FlowTargetSelect Component', () => {
+  const TestFlowTargetId = 'TestFlowTargetId';
+
+  const mockProps = {
+    id: TestFlowTargetId,
+    selectedDirection: FlowDirection.uniDirectional,
+    isLoading: false,
+    selectedTarget: FlowTarget.source,
+    updateFlowTargetAction: (jest.fn() as unknown) as ActionCreator<{
```

Comment:
@FrankHassanabad Any idea around this `unknown` typing? Was reading into testing w/ Typescript + Mocks [here](http://jonathancreamer.com/testing-typescript-classes-with-jest-and-jest-mocks/), but couldn't seem to find the right syntax when using an `ActionCreator`.

Reply:
this is how i got it to work

Reply:
Ditto. Looking for a better way myself :-)